packages/database/e2e/database.modular.e2e.js

```diff
@@ -0,0 +1,464 @@
+          return Promise.reject(new Error('Did not throw an Error.'));
+        } catch (error) {
+          error.message.should.containEql("'enabled' must be a boolean value");
+          return Promise.resolve();
+        }
+      });
+
+      it('calls setLoggingEnabled successfully', async function () {
+        firebase.database().setLoggingEnabled(true);
+        firebase.database().setLoggingEnabled(false);
+      });
+    });
+
+    describe('setPersistenceCacheSizeBytes()', function () {
+      it('throws if bytes is not a number', async function () {
+        try {
+          firebase.database().setPersistenceCacheSizeBytes('foo');
+          return Promise.reject(new Error('Did not throw an Error.'));
+        } catch (error) {
+          error.message.should.containEql("'bytes' must be a number value");
+          return Promise.resolve();
+        }
+      });
+
+      it('throws if bytes is less than 1MB', async function () {
+        try {
+          firebase.database().setPersistenceCacheSizeBytes(1234);
+          return Promise.reject(new Error('Did not throw an Error.'));
+        } catch (error) {
+          error.message.should.containEql("'bytes' must be greater than 1048576 bytes (1MB)");
+          return Promise.resolve();
+        }
+      });
+
+      it('throws if bytes is greater than 10MB', async function () {
+        try {
+          firebase.database().setPersistenceCacheSizeBytes(100000000000000);
+          return Promise.reject(new Error('Did not throw an Error.'));
+        } catch (error) {
+          error.message.should.containEql("'bytes' must be less than 104857600 bytes (100MB)");
+          return Promise.resolve();
+        }
+      });
+
+      it('calls setPersistenceCacheSizeBytes successfully', async function () {
+        firebase.database().setPersistenceCacheSizeBytes(1048576); // 1mb
+      });
+    });
+
+    describe('getServerTime()', function () {
+      it('returns a valid date', async function () {
+        const date = firebase.database().getServerTime();
+        date.getDate.should.be.Function();
+      });
+    });
+
+    describe('handles invalid references()', function () {
+      it('returns a valid date', async function () {
+        try {
+          firebase.database().ref('$');
+          return Promise.reject(new Error('Did not throw an Error.'));
+        } catch (error) {
+          error.message.should.containEql(
+            "Paths must be non-empty strings and can't contain #, $, [, ], ' or ?",
+          );
+          return Promise.resolve();
+        }
+      });
+    });
+  });
+
+  describe('modular', function () {
+    describe('namespace', function () {
+      it('accessible from getDatabase', function () {
+        const { getDatabase } = databaseModular;
+
+        const app = firebase.app();
+        const database = getDatabase();
+        should.exist(app.database);
+        database.app.should.eql(app);
```

Comment:
This doesn't look migrated, it doesn't use the app instance

Reply:
Could you elaborate on this? Do you mean that line 220 should be `const database = getDatabase(app);`?

Reply:
I think it is fine, the app property is available on the `getDatabase()` instance once initialised.

Reply:
Can this be resolved?